presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -265,6 +277,23 @@ private PlanBuilder appendExistSubqueryApplyNode(PlanBuilder subPlan, ExistsPred
         return appendSubqueryApplyNode(subPlan, existsPredicate, existsPredicate.getSubquery(), subqueryPlan, correlationAllowed);
     }
 
+    private PlanBuilder appendQuantifiedComparisonApplyNodes(PlanBuilder subPlan, Set<QuantifiedComparisonExpression> quantifiedComparisons, boolean correlationAllowed)
+    {
+        for (QuantifiedComparisonExpression quantifiedComparison : quantifiedComparisons) {
+            subPlan = appendQuantifiedComparisonApplyNode(subPlan, quantifiedComparison, correlationAllowed);
+        }
+        return subPlan;
+    }
+
+    private PlanBuilder appendQuantifiedComparisonApplyNode(PlanBuilder subPlan, QuantifiedComparisonExpression quantifiedComparison, boolean correlationAllowed)
+    {
+        if (subPlan.canTranslate(quantifiedComparison)) {
+            // given subquery is already appended
+            return subPlan;
+        }
+        throw new SemanticException(NOT_SUPPORTED, quantifiedComparison, "Quantified comparison with the given quantifier and comparison type is not supported");
```

Comment:
I'd rephrase the error message such that it's obvious to the user what the problem is. Specifically, include the operator and quantifier. I might even go with something like:

`">= ALL (...)" expressions not yet supported`

or something of the sort.